spec/models/bank_account_spec.rb

```diff
@@ -34,4 +36,46 @@
     subject { bank_account.complete_account_string }
     it { is_expected.to eq("#{bank_account.account} - #{bank_account.account_digit}") }
   end
+
+  describe "#input_bank_number_validation" do
+    before do
+      custom_bank
+    end
+
+    context "when input_bank_number has a invalid bank number" do
+      before do
+        bank_account.input_bank_number = "009123"
+      end
+
+      it { expect(bank_account.valid?).to be_falsey }
+    end
+
+    context "when input_bank_number has a valid bank" do
+      before do
+        bank_account.input_bank_number = "00M"
+      end
+      it { is_expected.to be_truthy }
+    end
+  end
+
+  describe "#load_bank_from_input_bank_number" do
+    before do
+      custom_bank
+    end
+
+    context "should be called when use .valid?" do
+      before do
+        expect(bank_account).to receive(:load_bank_from_input_bank_number)
+      end
+
+      it { bank_account.valid?}
+    end
+
+    it do
```

Comment:
We need a description here, could be: "should assign bank according to bank number"